evalml/data_checks/default_data_checks.py

```diff
@@ -19,11 +20,12 @@ class DefaultDataChecks(DataChecks):
         - `InvalidTargetDataCheck`
         - `NoVarianceDataCheck`
         - `ClassImbalanceDataCheck` (for classification problem types)
+        - `DateTimeNaNDataCheck`
 
     """
 
     _DEFAULT_DATA_CHECK_CLASSES = [HighlyNullDataCheck, IDColumnsDataCheck,
```

Comment:
Since AutoMLSearch will error out if a user provides datetime columns and has NaN values, I opted to add to default data checks as a safe guard.

Reply:
It seems like we don't have data checks as part of AutoMLSearch now. Open to removing this as well.

Reply:
I think it's fine to leave in for now, as we can still use the default data checks with `DefaultDataChecks().validate(X_train, y_train)`. As our data check process updates, this might change, but I see no issues in adding it here